framework/auth/campaigns.py

```diff
@@ -50,30 +45,27 @@ def get_campaigns():
         })
 
         # Proxy campaigns: Preprints, both OSF and branded ones
-        try:
-            preprint_providers = PreprintProvider.find(Q('_id', 'ne', None))
-            for provider in preprint_providers:
-                if provider._id == 'osf':
-                    template = 'osf'
-                    name = 'OSF'
-                    url_path = 'preprints/'
-                else:
-                    template = 'branded'
-                    name = provider.name
-                    url_path = 'preprints/{}'.format(provider._id)
-                campaign = '{}-preprints'.format(provider._id)
-                system_tag = '{}_preprints'.format(provider._id)
-                CAMPAIGNS.update({
-                    campaign: {
-                        'system_tag': system_tag,
-                        'redirect_url': furl.furl(DOMAIN).add(path=url_path).url,
-                        'confirmation_email_template': mails.CONFIRM_EMAIL_PREPRINTS(template, name),
-                        'login_type': 'proxy',
-                        'provider': name,
-                    }
-                })
-        except (NoResultsFound or QueryException or ImproperConfigurationError) as e:
-            logger.warn('An error has occurred during campaign initialization: {}', e)
+        preprint_providers = PreprintProvider.objects.all()
```

Comment:
`try ...  except ..` is removed. `.objects.all()` will return empty if no provider is found, which is safe for the `for` loop below. No exception will be thrown.